tests/integration/helpers/project_builder.rs

```diff
@@ -7,23 +7,40 @@ use std::sync::atomic::*;
 
 use crate::helpers::project::Project;
 use remove_dir_all::remove_dir_all;
+use tempfile::{tempdir, TempDir};
 
 static CNT: AtomicUsize = AtomicUsize::new(0);
 thread_local!(static IDX: usize = CNT.fetch_add(1, Ordering::SeqCst));
 
 pub struct ProjectBuilder {
     files: Vec<(String, String)>,
     submodules: Vec<(String, String)>,
-    root: PathBuf,
+    root: RootType,
     git: bool,
     branch: Option<String>,
 }
 
+enum RootType {
```

Comment:
I like the idea so much, that I would vote for not introducing this enum, and change `root: RootType` to `root: TempDir` so that we can get rid of the self made tempdir thing that was stored in root before. 

I've checked it locally and then also `impl Drop for Project` should be removed.

Reply:
Then `fn root()` would shrink to something like 

```rust
fn root(name: &str) -> TempDir {
    tempdir().unwrap()
}
```

Reply:
this would also make some of the changes below obsolete.

Reply:
I also like the idea of just using tempdir, so I went along and cleaned it up - no more 
```rust
fn dir(name: &str) -> ProjectBuilder
```
either... hope its ok.